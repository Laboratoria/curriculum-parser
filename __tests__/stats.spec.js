const { computeCourseStats } = require('../lib/stats');


describe('stats.computeCourseStats', () => {
  it('should be a function', () => expect(typeof computeCourseStats).toBe('function'));

  it('should compute course duration based on parts duration', () => {
    const { stats } = computeCourseStats({
      syllabus: {
        foo: {
          parts: {
            '01-intro': { duration: 60 },
          },
        },
        bar: {
          parts: {
            '00-opening': { duration: 20 },
            '00-closing': { duration: 30 },
          },
        },
      },
    });

    expect(stats).toMatchSnapshot();
  });

  it('should not add decimals to durationString when decimal is 0', () => {
    const { stats } = computeCourseStats({
      syllabus: {
        foo: {
          parts: {
            '01-intro': { duration: 60 },
          },
        },
        bar: {
          parts: {
            '00-opening': { duration: 30 },
            '00-closing': { duration: 30 },
          },
        },
      },
    });

    expect(stats).toMatchSnapshot();
  });

  it('should count exercises', () => {
    const { stats } = computeCourseStats({
      syllabus: {
        foo: {
          parts: {
            '01-intro': { duration: 30 },
          },
        },
        bar: {
          parts: {
            '00-opening': { duration: 5 },
            '01-practice': {
              duration: 30,
              exercises: {
                '01-foo': {},
              },
            },
            '02-closing': { duration: 5 },
          },
        },
      },
    });

    expect(stats).toMatchSnapshot();
  });

  it('should not complain when unit has no parts?', () => {
    const { stats } = computeCourseStats({
      syllabus: {
        foo: {},
      },
    });

    expect(stats).toMatchSnapshot();
  });
});
