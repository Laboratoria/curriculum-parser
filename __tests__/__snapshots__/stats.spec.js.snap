// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stats.computeCourseStats should compute course duration based on parts duration 1`] = `
Object {
  "duration": 110,
  "durationString": "1.8h",
  "exerciseCount": 0,
  "partCount": 3,
  "unitCount": 2,
}
`;

exports[`stats.computeCourseStats should count exercises 1`] = `
Object {
  "duration": 70,
  "durationString": "1.2h",
  "exerciseCount": 1,
  "partCount": 4,
  "unitCount": 2,
}
`;

exports[`stats.computeCourseStats should not add decimals to durationString when decimal is 0 1`] = `
Object {
  "duration": 120,
  "durationString": "2h",
  "exerciseCount": 0,
  "partCount": 3,
  "unitCount": 2,
}
`;

exports[`stats.computeCourseStats should not complain when unit has no parts? 1`] = `
Object {
  "duration": 0,
  "durationString": "0min",
  "exerciseCount": 0,
  "partCount": 0,
  "unitCount": 1,
}
`;
